lib/Doctrine/ORM/Query/AST/JoinAssociationPathExpression.php

```diff
@@ -26,12 +26,4 @@ public function __construct($identificationVariable, $associationField)
         $this->identificationVariable = $identificationVariable;
         $this->associationField       = $associationField;
     }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function dispatch($sqlWalker)
-    {
-        return $sqlWalker->walkPathExpression($this);
-    }
```

Comment:
I've removed this method because there's no way a call to this method could've returned anything. The parent class contains an implementation of that method that raises a proper exception, so it seems just right to fall back to that.